packages/react-widgets/src/hooks/useWidgetSource.js

```diff
@@ -0,0 +1,29 @@
+import { _getApplicableFilters as getApplicableFilters } from '@carto/react-core/';
+import { selectSourceById } from '@carto/react-redux/';
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import useCustomCompareMemo from './useCustomCompareMemo';
+import { dequal as deepEqual } from 'dequal';
+
+/**
+ * Hook to obtain widget's source excluding the filters like in useSourceFilters
+ * @param  {object} props
+ * @param  {string} props.dataSource - ID of the source to get the filters from.
+ * @param  {string} props.id - ID of the widget that apply the filter you want to exclude.
+ */
+export default function useWidgetSource({ dataSource, id }) {
+  const rawSource = useSelector((state) => selectSourceById(state, dataSource));
+
+  const applicableFilters = useMemo(
+    () => getApplicableFilters(rawSource?.filters, id),
+    [rawSource?.filters, id]
+  );
+
+  return useCustomCompareMemo(
+    rawSource && {
+      ...rawSource,
```

Comment:
Why the ...rawSource here? Do you need other props, beyond filters, to be considered in comparison?